### ISSUE TITLE
Optimization suggestion to get rid of TCP_CORK

### ISSUE THREAD

**Issue by neeraj9** [monkey/monkey#205]
I notice that you are using TCP_CORK on/off in various places. After spending some time looking at the static file serving it appears unnecessary. I suggest that you use sendto(2) interface with MSG_MORE instead (see http://linux.die.net/man/2/sendto for details). Note that this will save two system calls to setsockopt for TCP_CORK in you case while serving every request.

I can work on a patch as well but it'l take some time since I just started going through the project.

Quote from man sendto(2)

> The caller has more data to send. This flag is used with TCP sockets to obtain the same
> effect as the TCP_CORK socket option (see tcp(7)), with the difference that this flag can
> be set on a per-call basis.

Disclaimer: I wrote my webserver and optimized to my hearts content only to realize that its missing a number of features. Now I am hunting for exiting projects to contribute instead. I found your project interesting (after evaluating quite a few open source available right now) and would like to contribute.

**Comment by janus** [comment]
@neeraj9  I would like to check out your webserver source code .

**Comment by edsiper** [comment]
Interesting approach to get rid of TCP_CORK. 

I am thinking how do we can handle that from the caller instead the network layer, as you can see network operations are handled by liana and mbedtls plugins, so for a TLS session case I am not sure what would be optimal. 

Let's give it a try, but I think send(2) is enough as it handle the required flag.

**Comment by neeraj9** [comment]
Yes. sendto() works for udp as well, but you deal with tcp only.

**Comment by neeraj9** [comment]
@janus I was planning to push it out but then it needs a lot of improvement even for a first release. I will think about it, but not today.

**Comment by janus** [comment]
@Neeraj.  Notify me when that is done.

On Fri, May 29, 2015 at 6:40 AM, Neeraj <notifications@github.com> wrote:

> @janus <https://github.com/janus> I was planning to push it out but then
> it needs a lot of improvement even for a first release. I will think about
> it, but not today.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/monkey/monkey/issues/205#issuecomment-106696436>.
>



-- 
P.S Please join our groups*:  *nigeriaarduinogroup@googlegroups.com
* or *jifunze-kufikiria@googlegroups.com  These are platforms for learning
and sharing  of knowledge.
                                     www.satajanus.com | *Satajanus  Nig.
Ltd*
